bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/PreviewServlet.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ *
+ *    Copyright 2020 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+package com.adobe.cq.commerce.core.components.internal.servlets;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.servlet.Servlet;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.commerce.core.components.services.UrlProvider;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.WCMMode;
+
+@Component(
+    service = Servlet.class,
+    immediate = true,
+    property = {
+        "sling.servlet.methods=GET",
+        "sling.servlet.paths=/bin/wcm/cif",
+        "sling.servlet.selectors=" + PreviewServlet.PREVIEW_PRODUCT + ", " + PreviewServlet.PREVIEW_CATEGORY,
+        "sling.servlet.extensions=html"
+    })
+public class PreviewServlet extends SlingSafeMethodsServlet {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PreviewServlet.class);
+
+    private static final String REFERER_HEADER = "Referer";
```

Comment:
Sorry I missed this one: I'd rather use `org.apache.http.HttpHeaders.REFERER` instead of introducing our own constant.

Reply:
fixed. thanks